downstairs/src/complete_jobs.rs

```diff
@@ -0,0 +1,55 @@
+use crate::JobId;
+
+/// Stores a flush operation and a set of complete jobs
+///
+/// The flush operation is not included in the list of complete jobs, but acts
+/// as a lower bound for dependencies; any job older than the flush is assumed
+/// to have completed.
+#[derive(Debug)]
+pub struct CompletedJobs {
+    last_flush: Option<JobId>,
+    completed: Vec<JobId>,
+}
+
+impl CompletedJobs {
+    pub fn new(last_flush: Option<JobId>) -> Self {
```

Comment:
is the last_flush argument always going to be `None`? we could drop the argument if so

Reply:
Not quite, it's `Some(..)` when we've received `Message::LastFlush` during negotiation!